src/platform/packages/shared/kbn-scout/src/playwright/page_objects/discover_app.ts

```diff
@@ -47,4 +48,123 @@ export class DiscoverApp {
+  }
+
+  async getHitCountInt(): Promise<number> {
+    const hitCount = await this.getHitCount();
+    return parseInt(hitCount.replace(/,/g, ''), 10);
+  }
+
+  async getChartTimespan(): Promise<string> {
+    // Wait until the attribute no longer contains "Loading"
+    const element = this.page.testSubj.locator('unifiedHistogramChart');
+    await expect(element).not.toHaveAttribute('data-time-range', /Loading/);
+
+    return (await element.getAttribute('data-time-range')) ?? '';
+  }
+
+  async clickHistogramBar() {
+    const canvas = this.page.locator('[data-test-subj="unifiedHistogramChart"] canvas');
+    await canvas.waitFor({ state: 'visible', timeout: 5000 });
+    // Click at the center of the canvas
+    await canvas.click();
+  }
+
+  async waitUntilSearchingHasFinished() {
+    await this.page.testSubj.waitForSelector('loadingSpinner', {
+      state: 'hidden',
+      timeout: 30000,
+    });
+  }
+
+  async getDocTableIndex(index: number): Promise<string> {
+    const rowIndex = index - 1; // Convert to 0-based index
+    const row = this.page.locator(`[data-grid-row-index="${rowIndex}"]`);
+    await row.waitFor({ state: 'visible' });
+    return await row.innerText();
+  }
+
+  async getDocTableField(index: number): Promise<string> {
+    const rowIndex = index - 1;
+    await this.page.testSubj.click('dataGridFullScreenButton');
+    const row = this.page.locator(`[data-grid-row-index="${rowIndex}"]`);
+    const text = await row.innerText();
+    await this.page.testSubj.click('dataGridFullScreenButton');
+    return text.trim();
+  }
+
+  async getChartInterval(): Promise<string> {
+    const button = this.page.testSubj.locator('unifiedHistogramTimeIntervalSelectorButton');
+    return (await button.getAttribute('data-selected-value')) || '';
+  }
+
+  async hasNoResults(): Promise<boolean> {
+    return await this.page.testSubj.isVisible('discoverNoResults');
+  }
+
+  async hasNoResultsTimepicker(): Promise<boolean> {
+    return await this.page.testSubj.isVisible('discoverNoResultsTimefilter');
+  }
+
+  async expandTimeRangeAsSuggestedInNoResultsMessage() {
+    const button = this.page.testSubj.locator('discoverNoResultsViewAllMatches');
+    await button.waitFor({ state: 'visible' });
+    await button.click();
+    await this.waitUntilSearchingHasFinished();
+  }
+
+  async revertUnsavedChanges() {
+    await this.page.testSubj.hover('unsavedChangesBadge');
+    await this.page.testSubj.click('unsavedChangesBadge');
+    await this.page.testSubj.waitForSelector('unsavedChangesBadgeMenuPanel', { state: 'visible' });
+    await this.page.testSubj.click('revertUnsavedChangesButton');
+    await this.waitUntilSearchingHasFinished();
+  }
+
+  async clickFieldSort(field: string, sortOption: string) {
+    const header = this.page.locator(`[data-test-subj="dataGridHeaderCell-${field}"]`);
+    await header.click();
+    await this.page.locator(`button:has-text("${sortOption}")`).click();
```

Comment:
2 clicks, but no state check between them. Let's check for UI to be ready/updated before the 2nd click